ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/LedgerPeers.hs

```diff
@@ -342,6 +470,31 @@ prop_accBigPoolStake  (LedgerPools lps@(_:_)) =
   where
     accumulatedStakeMap = accBigPoolStakeMap lps
 
+prop_recomputeRelativeStake :: LedgerPools -> Property
+prop_recomputeRelativeStake (LedgerPools []) = property True
+prop_recomputeRelativeStake (LedgerPools lps) = property $ do
+  lpk <- genLedgerPeersKind
+  let (accStake, relayAccessPointsUnchangedNonNegativeStake) = go (reStake lpk) lps (0, True)
+  return $     counterexample "recomputeRelativeStake: relays modified or negative pool stake calculated"
+                             relayAccessPointsUnchangedNonNegativeStake
+          .&&. accStake === 1
+          .&&. counterexample "violates idempotency"
+                              ((recomputeRelativeStake BigLedgerPeers . recomputeRelativeStake BigLedgerPeers $ lps) == recomputeRelativeStake BigLedgerPeers lps)
+  where
+    genLedgerPeersKind = elements [AllLedgerPeers, BigLedgerPeers]
+    reStake lpk = recomputeRelativeStake lpk lps
+    -- compare relay access points in both lists for equality
+    -- where we assume that recomputerelativestake doesn't change
+    -- the order, and sum up relative stake to make sure it adds up to 1
+    go ((normPoolStake, raps):rest) ((_, raps'):rest') (accStake, _) =
+      if raps == raps' && normPoolStake >= 0
+      then go rest rest' (accStake + normPoolStake, True)
+      else (accStake + normPoolStake, False)
+    go [] (_:_) (accStake, _) = (accStake, False)
+    go (_:_) [] (accStake, _) = (accStake, False)
+    go _ _ (accStake, relayAccessPointsUnchangedNonNegativeStake) = (accStake, relayAccessPointsUnchangedNonNegativeStake)
```

Comment:
What does this test? why was it added?

Reply:
This is used when we compute big ledger peers, I wanted to increase the confidence that they are correct for Genesis ofc, and so some simple properties I came up with are:
1. The accumulated relative stake adds up to unity
2. No pool relative stake can be less than 0
3. The relays aren't mangled
4. Running this function multiple times always produces the same result

Reply:
I would add some haddocks explaining so its easier to understand

Reply:
I've added a comment.

Reply:
If we encode `Rational`s as `Scientific` values, we'll lose the property that they add up to `1`, but maybe that's fine.

Reply:
We encode them as `Doubles`, which we have already been using internally.